specification/vmwarevirtustream/resource-manager/Microsoft.VMwareVirtustream/preview/2019-08-09-preview/vmwarevirtustream.json

```diff
@@ -0,0 +1,1174 @@
+      "description": "The properties of a private cloud resource",
+      "properties": {
+        "provisioningState": {
+          "description": "The provisioning state",
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Cancelled",
+            "Pending",
+            "Building",
+            "Updating"
+          ],
+          "x-ms-enum": {
+            "name": "PrivateCloudProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "circuit": {
+          "description": "An ExpressRoute Circuit",
+          "$ref": "#/definitions/Circuit"
+        },
+        "cluster": {
+          "description": "The default cluster used for management",
+          "$ref": "#/definitions/DefaultClusterProperties"
+        },
+        "clusters": {
+          "description": "The clusters",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "type": "string"
+          }
+        },
+        "endpoints": {
+          "description": "The endpoints",
+          "readOnly": true,
+          "$ref": "#/definitions/Endpoints"
+        },
+        "internet": {
+          "description": "Connectivity to internet is enabled or disabled",
+          "type": "string",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "name": "InternetEnum",
+            "modelAsString": true
+          }
+        },
+        "identitySources": {
+          "description": "vCenter Single Sign On Identity Sources",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IdentitySource"
+          }
+        },
+        "networkBlock": {
+          "description": "The block of addresses should be unique across VNet in your subscription as well as on-premise. Make sure the CIDR format is conformed to (A.B.C.D/X) where A,B,C,D are between 0 and 255, and X is between 0 and 22",
+          "type": "string"
+        },
+        "managementNetwork": {
+          "description": "Network used to access vCenter Server and NSX-T Manager",
+          "type": "string",
+          "readOnly": true
+        },
+        "provisioningNetwork": {
+          "description": "Used for virtual machine cold migration, cloning, and snapshot migration",
+          "type": "string",
+          "readOnly": true
+        },
+        "vmotionNetwork": {
+          "description": "Used for live migration of virtual machines",
+          "type": "string",
+          "readOnly": true
+        },
+        "vcenterPassword": {
+          "description": "Optionally, set the vCenter admin password when the private cloud is created",
+          "type": "string"
+        },
+        "nsxtPassword": {
+          "description": "Optionally, set the NSX-T Manager password when the private cloud is created",
+          "type": "string"
+        }
```

Comment:
Any password or secret property should be annotated with x-ms-secret: true

Reply:
@majastrz, I don't see any specs using `x-ms-secret`. There are several using `"secret": true` and I also see `"isSecret": true`. Should I be using one of those?

Reply:
`x-ms-secret` was added recently and is not yet documented. Take a look at this query for examples: https://github.com/Azure/azure-rest-api-specs/search?q=x-ms-secret&unscoped_q=x-ms-secret

Reply:
Thanks. I added `"x-ms-secret": true`.

Reply:
Adding this leads to validation errors of:
```
Secret property `"nsxtPassword": "$(1X4Dkk"`, cannot be sent in the response.
```
Let'd put these changes in API 2020-03-20 as well. I'm going to revert the change.

Reply:
@MyronFanQiu this looks like a linter rule bug. Sharing definitions between GET and PUT is one of the accepted patterns and should not block usage of x-ms-secret: true. @ctaggart I'm ok either way, but you should be able to put it back in with a linter rule suppression as a workaround.

Reply:
Let's put in in API 2020-03-20.